minimum_jerk_trajectory_planner/include/minimum_jerk_trajectory_planner/robot.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include "trajectory_planners.hpp"
+#include "trajectory.hpp"
+#include "pose.hpp"
+#include <string.h>
+#include<memory> 
+
+
+namespace minimum_jerk
+{
+    class Robot
+    {
+    public:
+        char *name;
+        TrajectoryPlanner path_finder_controller;
+        double total_time;
+        Pose& pose;
+        Pose& pose_start;
```

Comment:
private, either vanilla `Pose` or a pointer to `Pose` (smart pointers, always), same for target